mlrun/api/crud/model_monitoring/stores/sql_model_endpoint_store.py

```diff
@@ -0,0 +1,609 @@
+
+        # Apply AND operator on the SQL query object with the filters tuple
+        return query.filter(db.and_(*filter_query))
+
+    @staticmethod
+    def _validate_labels(
+        endpoint_dict: dict,
+        labels: typing.Union[typing.List, typing.Dict[str, typing.Any]],
+    ) -> bool:
+        """Validate that the model endpoint dictionary has the provided labels. There are 3 possible cases:
+        1 - Labels were provided as a dictionary (e.g. {'label_1': 'value_1', 'label_2': 'value_2'}: Validate that the
+            labels dictionary is a subset of the endpoint labels dictionary.
+        2 - Labels were provided as a list of key-values pairs (e.g. ['label_1=value_1', 'label_2=value_2']): Validate
+            that each pair exist in the endpoint dictionary.
+        3 - Labels were provided as a list of key labels (e.g. ['label_1', 'label_2']): Validate that each key exist in
+            the endpoint labels dictionary.
+
+        :param endpoint_dict: Dictionary of the model endpoint records.
+        :param labels:        List of dictionary of required labels.
+
+        :return: True if the labels exist in the endpoint labels dictionary, otherwise False.
+        """
+
+        # Convert endpoint labels into dictionary
+        endpoint_labels = json.loads(
+            endpoint_dict.get(model_monitoring_constants.EventFieldType.LABELS)
+        )
+
+        # Case 1 - labels are dictionary
+        if (
+            isinstance(labels, typing.Dict)
+            and not labels.items() <= endpoint_labels.items()
+        ):
+            return False
+        for label in labels:
+            # Case 2 - labels are a list of key-value pairs
+            if "=" in label:
+                lbl, value = list(map(lambda x: x.strip(), label.split("=")))
+                if lbl not in endpoint_labels or str(endpoint_labels[lbl]) != value:
+                    return False
+            # Case 3 - labels are a list of keys
+            else:
+                if label not in endpoint_labels:
+                    return False
+
+        return True
+
+    def delete_model_endpoints_resources(
+        self, endpoints: mlrun.api.schemas.model_endpoints.ModelEndpointList
+    ):
+        """
+        Delete all model endpoints resources in both SQL and the time series DB. In addition, delete the model
+        endpoints table from SQL if it's empty at the end of the process.
+
+        :param endpoints: An object of `ModelEndpointList` which is literally a list of model endpoints along with some
+                          metadata. To get a standard list of model endpoints use `ModelEndpointList.endpoints`.
+        """
+
+        # Delete model endpoint record from SQL table
+        for endpoint in endpoints.endpoints:
+            self.delete_model_endpoint(
+                endpoint.metadata.uid,
+            )
+
+        # Drop the SQL table if it's empty
+        self._drop_table_if_not_empty()
```

Comment:
Do we really want to drop the table? This will mean it will not get re-created unless the class is initialized again.

Reply:
I think can keep it (see my comment here https://github.com/mlrun/mlrun/pull/2685#discussion_r1070628779) but if you think it might be dangerous or cause other issues then I'll remove this method.

Reply:
ok, so my thinking is that we should keep the table in pretty much all cases, even when deleting a project, since it's a single table that serves all projects. As for your comment - I agree that when doing changes to the format of the table it will require extra work, but dropping the table if it's empty doesn't really help you there since you may upgrade MLRun and the table still has data in it. This means that you'll need to do some sort of migration to the data anyway if the format changes. Best approach in your case, since you only have a single table, would probably be to change the name of the table (for example to `model_endpoints_v2`) and migrate any data in `model_endpoints` to the new table, then drop the old one (or, if it's not that critical, don't migrate data, just create new table and drop old one). However, this is something that we'll only need to worry about if and when there's any table format change.